plugins/inventory/opennebula.py

```diff
@@ -0,0 +1,225 @@
+                authenticate with ONE auth file. Default path is C(~/.one/one_auth).
+              - Set environment variable C(ONE_AUTH) to override this path.
+            env:
+              - name: ONE_AUTH
+            required: False
+            type: string
+        hostname:
+            description: Field to match the hostname. Note C(v4_first_ip) corresponds to the first IPv4 found on VM.
+            type: string
+            default: v4_first_ip
+            choices:
+                - v4_first_ip
+                - v6_first_ip
+                - name
+        filter_by_label:
+            description: Only return servers filtered by this label.
+            type: string
+'''
+
+EXAMPLES = r'''
+# inventory_opennebula.yml file in YAML format
+# Example command line: ansible-inventory --list -i inventory_opennebula.yml
+
+# Pass a label filter to the API
+plugin: community.general.opennebula
+api_url: https://opennebula:2633/RPC2
+filter_by_label: Cache
+'''
+
+try:
+    import pyone
+
+    HAS_PYONE = True
+except ImportError:
+    HAS_PYONE = False
+
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable
+from ansible.module_utils._text import to_native
+
+import os
+
+
+def _get_connection_info(url, username, password, authfile):
+    if not username:
+        if not password:
+            if authfile is None:
+                authfile = os.path.join(os.environ.get("HOME"), ".one", "one_auth")
+            try:
+                with open(authfile, "r") as fp:
+                    authstring = fp.read().rstrip()
+                username = authstring.split(":")[0]
+                password = authstring.split(":")[1]
+            except (OSError, IOError):
+                raise AnsibleError("Could not find or read ONE_AUTH file at '{e}'".format(e=authfile))
+            except Exception:
+                raise AnsibleError("Error occurs when read ONE_AUTH file at '{e}'".format(e=authfile))
+
+    from collections import namedtuple
+
+    auth_params = namedtuple('auth', ('url', 'username', 'password'))
+
+    return auth_params(url=url, username=username, password=password)
+
+
+def _get_vm_ipv4(vm):
+    nic = vm.TEMPLATE.get('NIC')
+
+    if isinstance(nic, dict):
+        nic = [nic]
+
+    for net in nic:
+        return net['IP']
+
+    return False
+
+
+def _get_vm_ipv6(vm):
+    nic = vm.TEMPLATE.get('NIC')
+
+    if isinstance(nic, dict):
+        nic = [nic]
+
+    for net in nic:
+        if net.get('IP6_GLOBAL'):
+            return net['IP6_GLOBAL']
+
+    return False
+
+
+def _retrieve_servers(one_client, label_filter=None):
+    # get hosts (VMs)
+    try:
+        vm_pool = one_client.vmpool.infoextended(-2, -1, -1, 3)
+    except Exception as e:
+        raise AnsibleError("Something happened during XML-RPC call, this was original exception: {e}".format(e=to_native(e)))
```

Comment:
It is quite a common pattern to rais e the exception and put the original one description in the message. I believe if we go
```suggestion
        raise AnsibleError("Something happened during XML-RPC call: {e}".format(e=to_native(e)))
```
it will be a bit more concise, but still clear enough for the user.